File: fix_prices.py
import re

files = {
    "src/app/dashboard/admin/listings/page.tsx": [
        ("formatPrice(selectedListing.price)", "selectedListing.price != null ? formatPrice(selectedListing.price) : '\u2014'"),
    ],
    "src/app/dashboard/admin/page.tsx": [
        ("formatPrice(l.price)", "l.price != null ? formatPrice(l.price) : '\u2014'"),
    ],
    "src/app/dashboard/admin/users/page.tsx": [
        ("formatPrice(l.price)", "l.price != null ? formatPrice(l.price) : '\u2014'"),
    ],
}

for filepath, replacements in files.items():
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()
    for old, new in replacements:
        content = content.replace(old, new)
    with open(filepath, "w", encoding="utf-8") as f:
        f.write(content)
    print(f"Fixed {filepath}")

# Fix recently-viewed.tsx - replace price display with category (homepage crash fix)
rv_path = "src/components/listings/recently-viewed.tsx"
with open(rv_path, "r", encoding="utf-8") as f:
    content = f.read()
content = content.replace(
    '<p className="text-blue-600 dark:text-blue-400 font-semibold text-xs mt-1">{formatPrice(item.price)}</p>',
    '<p className="text-blue-600 dark:text-blue-400 font-semibold text-xs mt-1 capitalize">{item.category}</p>'
)
with open(rv_path, "w", encoding="utf-8") as f:
    f.write(content)
print(f"Fixed {rv_path}")

# Remove unused formatPrice import from admin/favorites/page.tsx
fav_path = "src/app/dashboard/admin/favorites/page.tsx"
with open(fav_path, "r", encoding="utf-8") as f:
    content = f.read()
content = content.replace('import { formatPrice } from "@/lib/constants";\n', "")
with open(fav_path, "w", encoding="utf-8") as f:
    f.write(content)
print(f"Fixed {fav_path}")

print("All done!")

File: fix_prices.js
const fs = require('fs');

const fixes = [
  ['src/app/dashboard/admin/listings/page.tsx', 'formatPrice(selectedListing.price)', "selectedListing.price != null ? formatPrice(selectedListing.price) : '\u2014'"],
  ['src/app/dashboard/admin/page.tsx', 'formatPrice(l.price)', "l.price != null ? formatPrice(l.price) : '\u2014'"],
  ['src/app/dashboard/admin/users/page.tsx', 'formatPrice(l.price)', "l.price != null ? formatPrice(l.price) : '\u2014'"],
];

fixes.forEach(([file, from, to]) => {
  let c = fs.readFileSync(file, 'utf8');
  c = c.split(from).join(to);
  fs.writeFileSync(file, c, 'utf8');
  console.log('Fixed:', file);
});

// Fix recently-viewed.tsx - homepage crash (price display -> category)
let rv = fs.readFileSync('src/components/listings/recently-viewed.tsx', 'utf8');
rv = rv.replace('{formatPrice(item.price)}', '{item.category}');
fs.writeFileSync('src/components/listings/recently-viewed.tsx', rv, 'utf8');
console.log('Fixed: recently-viewed.tsx');

// Remove unused import from admin/favorites
let fav = fs.readFileSync('src/app/dashboard/admin/favorites/page.tsx', 'utf8');
fav = fav.replace('import { formatPrice } from "@/lib/constants";\n', '');
fav = fav.replace("import { formatPrice } from \"@/lib/constants\";\r\n", '');
fs.writeFileSync('src/app/dashboard/admin/favorites/page.tsx', fav, 'utf8');
console.log('Fixed: admin/favorites/page.tsx');

console.log('All done!');
